fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/lock/LockHolder.java

```diff
@@ -47,7 +47,7 @@ public int getRefCount() {
     }
 
     boolean isConflict(LockHolder lockHolderRequest) {
-        if (lockHolderRequest.locker == this.locker
+        if (lockHolderRequest.locker.equals(this.locker)
```

Comment:
feel like the condition should be `lockHolderRequest.locker.equals(this.locker) && !(this.lockType == LockType.READ && lockHolderRequest.lockType == LockType.WRITE)`